### PR TITLE
add startup fields to generator modification

### PR DESCRIPTION
**Pull request by anistouri**
Co-authored-by: anistouri <anis-1.touri@rte-france.com>

### CODE REVIEW

File: src/components/dialogs/generator-modification-dialog.js [hunk 1]
```diff
@@ -524,12 +524,54 @@ const GeneratorModificationDialog = ({
         defaultValue: getValue(formValues?.stepUpTransformerReactance),
         previousValue: generatorInfos?.stepUpTransformerReactance,
     });
+
+    const [plannedActivePowerSetPoint, plannedActivePowerSetPointField] =
+        useDoubleValue({
+            label: 'PlannedActivePowerSetPoint',
+            adornment: ActivePowerAdornment,
+            inputForm: inputForm,
```
Comment (flomillot) [inline src/components/dialogs/generator-modification-dialog.js:532]: Christelle asked me to add "optional" in the label, you have to do like here : https://github.com/gridsuite/gridstudy-app/pull/1047/files

File: src/components/dialogs/generator-modification-dialog.js [hunk 2]
```diff
@@ -690,7 +732,11 @@ const GeneratorModificationDialog = ({
             undefined,
             editData?.uuid,
             isVoltageRegulationOn && isDistantRegulation ? qPercent : null,
+            plannedActivePowerSetPoint ?? null,
+            startupCost ?? null,
             marginalCost ? marginalCost : null,
```
Comment (flomillot) [inline src/components/dialogs/generator-modification-dialog.js:737]: Maybe change this one too ?

**Review by SlimaneAmar — APPROVED** [review]

File: src/components/dialogs/generator-modification-dialog.js (hunk 1 above)
Comment (anistouri) [reply src/components/dialogs/generator-modification-dialog.js:532]: We have a ticket to remove "optional" in the label in the generator modification form
https://dev.azure.com/GridSuite/MVP/_sprints/backlog/MVP%20Team/MVP/Sprint%2058?workitem=1737

File: src/components/dialogs/generator-modification-dialog.js (hunk 2 above)
Comment (anistouri) [reply src/components/dialogs/generator-modification-dialog.js]: I did it

File: src/components/dialogs/generator-modification-dialog.js (hunk 1 above)
Comment (flomillot) [reply src/components/dialogs/generator-modification-dialog.js:532]: Ah yeah anyway all fields are optional in a modification form... my bad

**Review by flomillot — APPROVED** [review]

**Pull request merged**